src/app/models/form.server.model.ts

```diff
@@ -419,27 +422,31 @@ const compileFormModel = (db: Mongoose): IFormModel => {
     return newForm
   }
 
-  // Transfer ownership of the form to another user
-  FormSchema.methods.transferOwner = async function (
-    currentOwner: IUserSchema,
-    newOwnerEmail: string,
-  ) {
-    // Verify that the new owner exists
+  FormSchema.methods.transferOwner = async function (newOwnerEmail: string) {
+    const currentOwner = await User.findById(this.admin)
+    if (!currentOwner) {
+      throw new DatabaseError('Admin of the form cannot be found')
+    }
+    if (newOwnerEmail === currentOwner.email) {
+      throw new TransferOwnershipError('You are already the owner of this form')
+    }
+    // Verify that the new owner exists.
     const newOwner = await User.findOne({ email: newOwnerEmail })
-    if (newOwner == null) {
-      throw new Error(
+    if (!newOwner) {
+      throw new TransferOwnershipError(
         `${newOwnerEmail} must have logged in once before being added as Owner`,
       )
     }
 
-    // Update form's admin to new owner's id
+    // Update form's admin to new owner's id.
     this.admin = newOwner._id
 
-    // Remove new owner from perm list and include previous owner as an editor
-    this.permissionList = filter(this.permissionList, (item) => {
-      return item.email !== newOwnerEmail
-    })
+    // Remove new owner from perm list and include previous owner as an editor.
+    this.permissionList =
+      this.permissionList?.filter((item) => item.email !== newOwnerEmail) ?? []
```

Comment:
quick question - why is the optional chaining required? TS doesn't know about the type of `this`?

Reply:
permissionList is not required, since there is a default. But the mongoose TypeScript typings are not good enough to know defaults. This is just for typeguarding

Reply:
hmm is it safe to set it to `required`? if not we can stay the course

Reply:
will result in us needed to change every creation (technically only in tests) to have an explicit permissionList array. not worth it i think